src/olympia/devhub/views.py

```diff
@@ -608,13 +608,10 @@ def check_addon_compatibility(request):
                        None, None, request=request)})
 
 
-def handle_upload(filedata, user, app_id=None, version_id=None, addon=None,
-                  is_standalone=False, is_listed=True, automated=False,
-                  submit=False):
-    if addon:
-        # TODO: Handle betas.
-        automated = addon.automated_signing
-        is_listed = addon.is_listed
+def handle_upload(filedata, user, channel, app_id=None, version_id=None,
+                  addon=None, is_standalone=False, submit=False):
+    automated = channel == amo.RELEASE_CHANNEL_UNLISTED
```

Comment:
Can you change this variable name while you're at it ? It should be `automated_signing` or something similar to denote that it's used to... sign automatically.